### ISSUE TITLE
Leftover Console.log in Slick.Grid.js

### ISSUE THREAD

**Issue by ghiscoding** [6pac/SlickGrid#392]
@6pac 
I think you forgot some logging stuff in pushing your new version. I now see this under my repo unit tests... 😆 

```bash
PASS  src/app/modules/angular-slickgrid/components/angular-slickgrid.component.spec.ts
  ● Console

    console.log node_modules/slickgrid/slick.grid.js:2168
      Col Widths:
    console.log node_modules/slickgrid/slick.grid.js:2168
      Col Widths:
    console.log node_modules/slickgrid/slick.grid.js:2168
      Col Widths:
    console.log node_modules/slickgrid/slick.grid.js:2168
      Col Widths:
    console.log node_modules/slickgrid/slick.grid.js:2168
      Col Widths:
    console.log node_modules/slickgrid/slick.grid.js:2168
      Col Widths:
    console.log node_modules/slickgrid/slick.grid.js:2168
      Col Widths:
    console.log node_modules/slickgrid/slick.grid.js:2168
      Col Widths:
    console.log node_modules/slickgrid/slick.grid.js:2168
      Col Widths:
    console.log node_modules/slickgrid/slick.grid.js:2168
      Col Widths:
    console.log node_modules/slickgrid/slick.grid.js:2168
      Col Widths:
```

I'd be happy not to see them at all lol

It also looks like the latest version breaks my auto-resize feature, I'll have to review your Wiki for the new code. I thought you would have made the old code as the default but I assumed wrong. Quickly reading the Wiki seemed like an interesting new way of dealing with column size.

**Comment by 6pac** [comment]
Will remove the logs. Sorry, I wasn't aware you had an auto-resize feature! It's not in the main trunk, is it? I did preserve the legacy slickgrid behaviour by default.

**Comment by 6pac** [comment]
Also the new example is particularly awesome ;-) http://6pac.github.io/SlickGrid/examples/example-size-to-content.html

**Comment by ghiscoding** [comment]
No problem

As for the size issue that I have, I think it's because I call the `autosizeColumns` after I auto-resize the grid, and that might be where that is broken (I think you added some new arguments to that function thought I was calling it without). I never used the `forceFitColumns` because it had some weird behaviors and `autosizeColumns` was always my go to. So the new issue that I have with the new version is that the columns are not being resized and they no longer fit the screen. 

I did see your new crazy Example and the long Wiki you made, nice job. I'll have to sit down and take some moment to try it out. I do like how the columns are now more adapted to their content. 👍

**Comment by 6pac** [comment]
Hmmm, as far as I can tell, ```autosizeColumns``` should work as it did. 

The only thing is that ```options.forceFitColumns``` is required to tell the grid to set ```GridAutosizeColsMode.Legacy```  rather than the default of ```GridAutosizeColsMode.None```.

Do you call ```autosizeColumns()``` without using ```options.forceFitColumns```? If so, I didn't allow for that use case. I suppose I could default to ```GridAutosizeColsMode.Legacy``` and just not call ```autosizeColumns()``` on init unless  ```options.forceFitColumns``` is specified.

**Comment by 6pac** [comment]
BTW in the call signature ```function autosizeColumns(autosizeMode, isInit) {```  
```autosizeMode``` falls back to the value specified in ```options``` if not specified - this param is so the call can override the default mode without setting it
```isInit``` tells the algorithm that the call is part of the grid```init``` process, so it can use initial sort order in its calculations

**Comment by ghiscoding** [comment]
> Do you call autosizeColumns() without using options.forceFitColumns? If so, I didn't allow for that use case. I suppose I could default to GridAutosizeColsMode.Legacy and just not call autosizeColumns() on init unless options.forceFitColumns is specified.

Yeah that's the thing, I do disable `forceFitColumns` while I'm calling `autosizeColumns` because I never liked it's behavior. If I just need to pass `GridAutosizeColsMode.Legacy` as argument to get back to working, that's not a big deal, I can do that and pass it as an argument, I just didn't have time to try yet.

**Comment by 6pac** [comment]
Yep, it should be that simple - to be precise, pass ```Slick.GridAutosizeColsMode.Legacy```.

I'm still confused though - didn't ```forceFitColumns``` just call ```autosizeColumns``` anyway? I didn't realise there was a difference.

**Comment by ghiscoding** [comment]
It's a little hard to understand just by words, so let me add 2 animated GIFs with a brief explanation. The behavior I didn't like about the Force Fit is that it's (was) ON at all time, this meant that if I resize a column, it must take the space for the other column(s) and I didn't like (neither my users) like this approach. What I prefer is that, it does the Force Fit on page load but don't I don't want it to restrain me from over sizing afterward. With that in mind, I found that yes when I use the `autosizeColumns`, I do use it with `forceFitColumns` on page load, but I disable it right after, this allows to what is shown in the 2nd animated GIF

I believe, your new approach fixes the behavior that I always disliked. Is that correct?

with `forceFitColumns`, I'm stuck with it, it will always take get other columns smaller to get 1 bigger
![LmXDEkJ5ZE](https://user-images.githubusercontent.com/643976/61260741-61677000-a74d-11e9-95d7-e4ed25e2727e.gif)

The way I have it implemented in my repo, I use `forceFitColumns` only on page load, I disable it right after
![D0L31wmrbi](https://user-images.githubusercontent.com/643976/61260745-66c4ba80-a74d-11e9-8fdd-248439ec87c0.gif)

**Comment by 6pac** [comment]
Good to know. I'll have a think on the best approach, but will definitely allow for your use case.

**Comment by 6pac** [comment]
Had a look, actually I can't really imagine anyone would really want their columns widths controlled after every column resize. I'm tempted to make a breaking change and remove this behaviour (so the resize only occurs on init). Do you think that's acceptable?

**Comment by ghiscoding** [comment]
> Had a look, actually I can't really imagine anyone would really want their columns widths controlled after every column resize. 

That is why I never liked that behavior hehe

> I'm tempted to make a breaking change and remove this behaviour (so the resize only occurs on init). Do you think that's acceptable? 

I wouldn't mind myself since I always bypass it this behavior with the hack I explained earlier. However, if you are to make a breaking change, I would recommend a `2.5.x` to go with semantic versioning (semver). I think it should have been `2.5.x` as soon as you released this new column resize, but it's still ok to do it now.

**Comment by ghiscoding** [comment]
Ok so I just tried out your new code and I seem to have 2 problems

1. I use TypeScript in my repos and it seems that the `colDataTypeOf` of type Date and Moment are not valid. 

If I change your previous code from
```js
if (colDataTypeOf === 'date') {
  autoSize.colValueArray = [ new Date(2009, 08, 30, 12, 20, 20) ]; // Sep 30th 2009, 12:20:20 AM
}
if (colDataTypeOf === 'moment') {
  autoSize.colValueArray = [ moment([2009, 08, 20, 12, 20, 20]) ]; // Sep 30th 2009, 12:20:20 AM
}
```
to this 
```js
if (colDataTypeOf === 'date') {
  autoSize.colValueArray = [ new Date("September 30, 2009 12:20:30") ]; // Sep 30th 2009, 12:20:20 AM
}
if (colDataTypeOf === 'moment' && typeof moment === 'function') {
  autoSize.colValueArray = [ moment("2009-09-30 12:20:20") ]; // Sep 30th 2009, 12:20:20 AM
}
```
then I get no more errors, I'm not sure why you had the `moment` called with an array as argument? 

2. then after calling `autosizeColumns(Slick.GridAutosizeColsMode.Legacy)`, it resizes ok at first but then after trying to resize, I see that the horizontal scroll never shows up. I also see this problem in your new demo, so that seems to be after your new feature. See below for an animation of the problem, at some point I don't see the last column anymore and there's still no horizontal scroll

![qmxCaxxlew](https://user-images.githubusercontent.com/643976/61266696-ff196a00-a762-11e9-9e53-80cf24d5f26d.gif)

**Comment by ghiscoding** [comment]
@6pac 
I've created a PR to fix the parsing issue found while doing a build. 

However what is shown in the animated GIF, there is no more any horizontal scrolling, is still an issue and a really big one. I cannot upgrade to the latest because of that, it's a show stopper for me.

**Comment by ghiscoding** [comment]
@6pac
I found where the problem is (horizontal scrolling missing), but I'm not sure what is causing it. The height of one of the viewport is too high and the horizontal scrollbar is actually hidden behind it. If I resize the viewport by removing about 25px, I now see the horizontal scroll. See animated GIF for explanation.

![2019-07-17_09-47-50](https://user-images.githubusercontent.com/643976/61380944-48a7a900-a878-11e9-8404-643b36259efc.gif)

**Comment by ghiscoding** [comment]
@6pac 
I found the issue, it was caused by PR #375 (option to hide column names), there's a flag which needs to be set `showColumnLabels: true,` in the slickgrid default option.

However can you merge my other PR before I fix this one.

**Comment by 6pac** [comment]
This is really weird. It looks like part of that PR has gone through, and the rest has been reverted or overwritten. That's totally unexpected and a bit worrying. I'll have to go back through the commit history. For now, I can fix that up if you like.

**Comment by ghiscoding** [comment]
@6pac 
oops I just pushed directly to `master`, I forgot to create a branch again... so it's fixed already lol

The only piece of code remaining is the `LogColWidths`, can that be removed or at least commented out? I see you are calling it in 2 places. Also BTW, I don't see any (visually) differences between `FitColsToViewport` and `FitViewportToCols`

**Comment by 6pac** [comment]
I don't think ```options.showColumnLabels``` should be an option. It looks as if this is just to make it accessible. Wouldn't it make more sense if it were an internal variable, and we offered ```getColumnHeaderVisibility```?

Try ```FitViewportToCols``` immediately after page load, before ```FitColsToViewport ```.  If you've already fit the cols to the viewport, then fitting the viewport to the cols does nothing ;-) 

In terms of the issues with legacy behaviour, I think an option like ```autoResizeGridColumnsAfterUserColumnResize``` (a bit wordy :-) would be the best way to deal with it. User-resized columns would have to be marked so they weren't auto resized, and I'd have to offer a way to reset user sized columns and go back to resizing everything.

**Comment by 6pac** [comment]
damn, were' getting in a twist over this one! Surely ```options.showColumnLabels``` should just be ```options.showColumnHeaders```! I think it was just a typo in the original PR...

**Comment by ghiscoding** [comment]
Nice twist, it does seems to work properly by renaming all the `showColumnLabels` to `showColumnHeaders` and the `example8-alternative-display.html` that was in that PR does seem to work too.

**Comment by ghiscoding** [comment]
I believe this was all done and fixed
